src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/bqsr/BaseRecalibratorFn.java

```diff
@@ -116,13 +89,20 @@ public void processElement(ProcessContext c) throws Exception {
             onTraversalStart();
             firstInBundle = false;
         }
-        GATKRead r = c.element().getKey();
+
+        GATKRead read = c.element().getKey();
+        if ( read instanceof SAMRecordToGATKReadAdapter ) {
+            // we ship those without the header, put it back.
+            SAMRecordToGATKReadAdapter readAdapter = (SAMRecordToGATKReadAdapter)read;
+            readAdapter.setHeader(header);
```

Comment:
I suspect this is unnecessary, as the tests were passing before and we weren't doing that. Perhaps the BaseRecal doesn't call any of the functions that needs a header?

At any rate, we need a utility function for that. No point in repeating this same code all over the place.